frontend/src/app/page.tsx

```diff
@@ -1,42 +1,109 @@
-    "root-objects-cid-v3",
-    []
-  );
+import { useCallback, useEffect, useState } from "react";
+import { isAuthenticated } from "../utils/auth";
 
-  const [rootObjectMetadata, setRootObjectMetadata] =
-    useState<UploadedObjectMetadata[]>();
+export default function App() {
+  const [isLoading, setIsLoading] = useState(false);
 
   useEffect(() => {
-    Promise.all(
-      localObjectCIDs.map((e) => ApiService.fetchUploadedObjectMetadata(e))
-    ).then(setRootObjectMetadata);
+    isAuthenticated().then((session) => {
+      if (session) {
+        window.location.assign("/drive");
+      }
+    });
+  }, []);
+
+  const handleAuth = useCallback(() => {
+    setIsLoading(true);
+
+    // Replace this URL with your actual Google OAuth 2.0 authorization endpoint
+    const authUrl = "https://accounts.google.com/o/oauth2/v2/auth";
+
+    // These are example parameters. Replace with your actual values
+    const params = new URLSearchParams({
+      client_id: process.env.NEXT_PUBLIC_GOOGLE_AUTH_CLIENT_ID!,
+      redirect_uri: "http://localhost:8080/api/auth/callback/google",
+      response_type: "code",
+      access_type: "offline",
+      prompt: "consent",
+      scope: "openid email",
+    });
+
+    // Redirect to the authorization URL
+    window.location.assign(`${authUrl}?${params.toString()}`);
   }, []);
 
   return (
-    <div className="flex w-full">
-      <div className="w-full flex flex-col gap-4">
-        <FileDropZone />
-        <div className="">
-          <UploadingObjects />
-          {rootObjectMetadata && rootObjectMetadata.length > 0 && (
-            <FileTable files={rootObjectMetadata} />
-          )}
-          {rootObjectMetadata && rootObjectMetadata.length === 0 && (
-            <NoUploadsPlaceholder />
-          )}
+    <div className="min-h-screen flex items-center justify-center bg-gradient-to-r from-blue-500 to-purple-600">
+      <div className="bg-white p-8 rounded-lg shadow-2xl w-full max-w-md">
+        <h1 className="text-3xl font-bold mb-6 text-center text-gray-800">
+          Welcome Back
+        </h1>
+        <p className="text-center mb-8 text-gray-600">
+          Sign in with your Google account to continue
+        </p>
+        <div className="flex justify-center">
+          <button
+            onClick={handleAuth}
+            disabled={isLoading}
+            className="bg-black hover:bg-gray-800 text-white font-bold py-3 px-6 rounded-full 
+                       border-2 border-transparent hover:border-white focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-gray-900
+                       transition duration-300 ease-in-out transform hover:-translate-y-1 hover:scale-105
+                       flex items-center justify-center w-full max-w-xs"
+            aria-label="Sign in with Google"
+          >
+            {isLoading ? (
+              <div className="flex items-center justify-center">
+                <svg
```

Comment:
Can we put svg either in a .svg or in a .tsx file with the name of what they are, to improve readability, please.